cpp/src/groupby/sort/sort_helper.cu

```diff
@@ -195,17 +189,16 @@ sort_groupby_helper::index_vector const& sort_groupby_helper::group_labels(
   if (_group_labels) return *_group_labels;
 
   // Get group labels for future use in segmented sorting
-  _group_labels = std::make_unique<index_vector>(num_keys(stream), stream);
+  auto group_labels = std::make_unique<index_vector>(num_keys(stream), stream);
```

Comment:
Sorry I forgot to mention about commenting for this too.
```suggestion
  // Create a temporary variable and only set _group_labels right before the return.
  // This way, a 2nd (parallel) call to this will not be given a partially created object.
  auto group_labels = std::make_unique<index_vector>(num_keys(stream), stream);
```